manifests/repo.pp

```diff
@@ -0,0 +1,15 @@
+# Configure the foreman repo
+class foreman::repo(
+  $custom_repo = $::foreman::custom_repo,
+  $repo        = $::foreman::repo,
+  $gpgcheck    = $::foreman::gpgcheck
+) {
+  if ! $custom_repo {
+    foreman::repos { 'foreman':
+      repo     => $repo,
+      gpgcheck => $gpgcheck,
+    }
+  }
+
+  include ::foreman::repos::extra
```

Comment:
I don't think this will get contained properly, you might need to add an anchor, use the contain keyword (and depend on Puppet >= 3.4(?)) or try `-> Class['foreman::repo']`.